plugins/cpp/parser/src/clangastvisitor.h

```diff
@@ -921,6 +940,32 @@ class ClangASTVisitor : public clang::RecursiveASTVisitor<ClangASTVisitor>
     return true;
   }
 
+  bool VisitStmt(clang::Stmt* s_)
+  {
+    if (clang::DeclStmt* ds = llvm::dyn_cast<clang::DeclStmt>(s_))
+    {
+      // FIXME: Using the parent map is expensive.
```

Comment:
getParentMapContext is said to be expesnive, and probably not the best approach. We should instead use some sort of statement cache to retrieve the parent of the current statement during traversal.